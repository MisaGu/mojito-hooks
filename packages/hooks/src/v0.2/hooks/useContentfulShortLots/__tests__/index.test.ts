import { act, renderHook } from '@testing-library/react-hooks';
import { TestWrapper } from '../../../components/test/wrapper/TestWrapper';
import {
  useContentfulShortLots,
  UseContentfulShortLotsProps,
  UseContentfulShortLotsReturn,
} from '../useContentfulShortLots';

describe('useContentfulShortLots()', () => {
  it('should be defined', () => {
    expect(useContentfulShortLots).toBeDefined();
  });

  it('returns the right result when needed', async () => {
    const { result, rerender, waitFor } = renderHook<
      [UseContentfulShortLotsProps],
      UseContentfulShortLotsReturn
    >(
      (args) => {
        return useContentfulShortLots(...args);
      },
      {
        initialProps: [
          {
            mojitoID: [
              '5f6a59f0-63bb-4628-b0ff-fcad26559a09',
              '600f19c6-0bad-4442-bc69-1d2db645292c',
            ],
          },
        ],
        wrapper: TestWrapper as any,
      },
    );

    // Trigger request...
    await act(async () => rerender());

    // ...and wait for response:
    await waitFor(() => result.current.queryResult.isSuccess);

    // Check data matches snapshot:
    expect(result.current).toMatchSnapshot({
      queryResult: {
        dataUpdatedAt: expect.any(Number),
      },
    });
  });
});
