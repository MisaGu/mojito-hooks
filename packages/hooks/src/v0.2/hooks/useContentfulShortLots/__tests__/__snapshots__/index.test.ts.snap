// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useContentfulShortLots() returns the right result when needed 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": true,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
  "shortLots": Array [
    Object {
      "author": Object {
        "avatar": Object {
          "url": "https://images.ctfassets.net/fu9did2d8yaw/p9Ci1Lw3OfeHfeHT2oysW/d32846a84de8d677fa155974dd610b27/960x0-683x1024.jpeg",
        },
        "name": "Mad Dog Jones",
        "slug": "mad-dog-jones",
      },
      "collector": Object {
        "avatar": Object {
          "url": "https://images.ctfassets.net/fu9did2d8yaw/fHmXsOiDpFTpyyoFt1adF/2bee40eb3668003edeff2355be320073/ND2_Collector_888.jpeg",
        },
        "name": "888",
        "slug": "888",
      },
      "gridPreviewImage": null,
      "imagesCollection": Object {
        "items": Array [
          Object {
            "title": "Thought as a System #6/50",
            "url": "https://images.ctfassets.net/fu9did2d8yaw/UBmFtd1jZIiCgXBLzq5Z1/d8aaba9955f5d1d6da9550eacec308b1/ND1.2_Thought-as-a-system_Preview.png",
          },
        ],
      },
      "lotId": 17,
      "mojitoId": "5f6a59f0-63bb-4628-b0ff-fcad26559a09",
      "slug": "thought_as_a_system_650",
      "subtitle": null,
      "sys": Object {
        "publishedAt": "2022-02-18T21:06:18.714Z",
      },
      "title": "Thought as a system #6/50",
    },
    Object {
      "author": Object {
        "avatar": Object {
          "url": "https://images.ctfassets.net/fu9did2d8yaw/314U28LCWpkad3CIcAndRC/be1cdd4130ae17fdccd4966cf482cb4e/ND2_Artist_Ponderware.png",
        },
        "name": "Ponderware",
        "slug": "ponderware",
      },
      "collector": Object {
        "avatar": Object {
          "url": "https://images.ctfassets.net/fu9did2d8yaw/fHmXsOiDpFTpyyoFt1adF/2bee40eb3668003edeff2355be320073/ND2_Collector_888.jpeg",
        },
        "name": "888",
        "slug": "888",
      },
      "gridPreviewImage": null,
      "imagesCollection": Object {
        "items": Array [
          Object {
            "title": "Wrapped MoonCats #3174",
            "url": "https://images.ctfassets.net/fu9did2d8yaw/5Br2Yp7PNWpTXGW7kqghp/ae66ffe412bf80fae1b670d4a7914080/wrappedMoonCats3174.png",
          },
        ],
      },
      "lotId": 28,
      "mojitoId": "600f19c6-0bad-4442-bc69-1d2db645292c",
      "slug": "wrapped_mooncats_3174",
      "subtitle": null,
      "sys": Object {
        "publishedAt": "2022-02-18T21:27:36.926Z",
      },
      "title": "Wrapped MoonCats #3174",
    },
  ],
}
`;
